src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformArtifactoryForm.tsx

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import {
+  Button,
+  ButtonVariation,
+  Color,
+  Formik,
+  FormInput,
+  Layout,
+  SelectOption,
+  StepProps,
+  Text,
+  useToaster
+} from '@wings-software/uicore'
+import React, { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { map } from 'lodash-es'
+import cx from 'classnames'
+import { FieldArray, Form } from 'formik'
+
+import { useStrings } from 'framework/strings'
+import { useGetArtifactsBuildsDetailsForArtifactory, useGetRepositoriesDetailsForArtifactory } from 'services/cd-ng'
+import {
+  formatInitialValues,
+  terraformArtifactorySchema,
+  tfArtifactoryFormInputNames,
+  getConnectorRef,
+  formatOnSubmitData
+} from './TerraformArtifactoryFormHelper'
+import type { PathInterface } from '../TerraformInterfaces'
+import stepCss from '@pipeline/components/PipelineSteps/Steps/Steps.module.scss'
+
+import css from './TerraformVarfile.module.scss'
+
+interface TFRemoteProps {
+  onSubmitCallBack: (data: any, prevStepData?: any) => void
+  isConfig: boolean
+  isTerraformPlan: boolean
+}
+
+export const TFArtifactoryForm: React.FC<StepProps<any> & TFRemoteProps> = ({
+  previousStep,
+  prevStepData,
+  onSubmitCallBack,
+  isConfig,
+  isTerraformPlan
+}) => {
+  const [selectedRepo, setSelectedRepo] = useState('')
+  const [connectorRepos, setConnectorRepos] = useState<SelectOption[]>()
+  const [filePathIndex, setFilePathIndex] = useState<{ index: number; path: string }>()
+  const [artifacts, setArtifacts] = useState<any>()
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232248)
```suggestion
  const [artifacts, setArtifacts] = useState<unknown>()
```